### ISSUE TITLE
lower performance

### ISSUE THREAD

**Issue by CandiceUIC** [AlanChou/Deep-Co-Training-for-Semi-Supervised-Image-Recognition#3]
Hey Alan,
Thanks for your great work. 
I ran your code on python 3.6 (you are using 3.5 but I don't see this is a problem), I could only get accuracy around 88%. I have also tried to add .detach() to perturbed example when creating pseudo label, result is still lower than 89%. 
I am wondering if there is any other tricks to get 89% acc?
Thanks again for your work.

Cheers

**Comment by AlanChou** [comment]
Hi Candice,

I didn't use any other tricks.
Sorry for the inconsistent performance. As you can see from the log, I did get around 89% but I don't remember if it's just lucky with a certain seed or it can consistently perform like that. The main goal for me to implement this paper is actually learning PyTorch. In fact, this is my very first PyTorch attempt so the code is little bit messy and buggy.  If you're interested in semi-supervised learning and want to get the SOTA results, I'll recommend you to check out "Unsupervised Data Augmentation", "MixMatch: A Holistic Approach to Semi-Supervised Learning", "Interpolation Consistency Training for Semi-Supervised Learning". 

I will come up with an update in a few days and hopefully solve the problem. Stay tuned to the repo if you're still interested in co-training.

**Comment by CandiceUIC** [comment]
I see. Thank you so much for your work, I will have another look at the code and let you know if I could improve the result.

**Comment by AlanChou** [comment]
Hi, 

I've modified the code a little bit and I still get similar results (89%) .
